msal/authority.py

```diff
@@ -71,6 +71,14 @@ def __init__(self, authority_url, http_client, validate_authority=True):
         if isinstance(authority_url, AuthorityBuilder):
             authority_url = str(authority_url)
         authority, self.instance, tenant = canonicalize(authority_url)
+
+        # extract authority port
+        parsedUrl = urlparse(authority_url)
```

Comment:
I think "authority" is already parsed url: https://github.com/sarathys/microsoft-authentication-library-for-python/blob/f4036a38be7d0136383390422b98e9fe67082b93/msal/authority.py#L143